### PR TITLE
Tweaks and fixes

### PR DESCRIPTION
**Pull request by SuicidalInsanity**
-Gun recoil now properly takes propellant mass into account
-Multiple fireanim support for single barrel weapons
-fix AIR-2 Genie manufacturer loc tag
-AIGUI minimum DefaultAlt now matches PAW
-rcsReductionFactor can now be set > 1 for making radar reflectors and similar
-Fix impulse for lasers

### CODE REVIEW

File: BDArmory/UI/BDArmoryAIGUI.cs [hunk 1]
```diff
@@ -342,7 +342,7 @@ void SetInputFields(Type AIType)
                         { "autoTuningAltitude", gameObject.AddComponent<NumericInputField>().Initialise(0, ActivePilot.autoTuningAltitude, 50, 5000) },
                         { "autoTuningSpeed", gameObject.AddComponent<NumericInputField>().Initialise(0, ActivePilot.autoTuningSpeed, 50, 800) },
 
-                        { "defaultAltitude", gameObject.AddComponent<NumericInputField>().Initialise(0, ActivePilot.defaultAltitude, 100, 15000) },
+                        { "defaultAltitude", gameObject.AddComponent<NumericInputField>().Initialise(0, ActivePilot.defaultAltitude, 50, 15000) },
```
Comment (BrettRyland) [inline BDArmory/UI/BDArmoryAIGUI.cs:345]: Many of these values should actually depend on the slider resolution setting. It's been on my to-do list for a while to have these get adjusted to use the right values when the slider changes.

File: BDArmory/Weapons/ModuleWeapon.cs [hunk 2]
```diff
@@ -2995,6 +3002,7 @@ void PlayFireAnim()
                     }
                     fireState[i].speed = fireAnimSpeed;
                     fireState[i].normalizedTime = Mathf.Repeat(fireState[i].normalizedTime, 1);
+                    Debug.Log("[BDArmory.ModuleWeapon]: playing Fire Anim, i = " + i + "; fire anim " + fireState[i].name);
```
Comment (BrettRyland) [inline BDArmory/Weapons/ModuleWeapon.cs:3005]: This should probably be behind a `if (BDArmorySettings.DEBUG_WEAPONS)` clause.

File: BDArmory/Weapons/ModuleWeapon.cs [hunk 3]
```diff
@@ -2854,6 +2863,7 @@ public void UpdateLaserSpecifics(bool newColor, bool newTex, bool newWidth, bool
                                 //Debug.Log("[BDArmory.ModuleWeapon]: barrelIndex for " + this.GetShortName() + " reset");
                             }
                         }
+                        if (animIndex + 1 > fireState.Length) animIndex = 0;
```
Comment (BrettRyland) [inline BDArmory/Weapons/ModuleWeapon.cs:2866]: `animIndex >= fireState.Length` is cleaner.

File: BDArmory/Weapons/ModuleWeapon.cs (hunk 2 above)
Comment (SuicidalInsanity) [reply BDArmory/Weapons/ModuleWeapon.cs]: Or commented out/deleted; there's always one debug message that I miss...

File: BDArmory/Weapons/ModuleWeapon.cs (hunk 3 above)
Comment (SuicidalInsanity) [reply BDArmory/Weapons/ModuleWeapon.cs]: yes, except it's dealing with a 0-based index, which means needing the +1 to convert to the 1-based firestate count; animIndex >= firestate.length would have it reset early.

Comment (BrettRyland) [reply BDArmory/Weapons/ModuleWeapon.cs]: Sorry, but I don't follow your logic here. `a + 1 > b` is the same as `a >= b` for integers, which is what you'd use if `animIndex` was being used as an index into `fireState`.
If `animIndex` is supposed to go up to `fireState.Length` before resetting (i.e., `0` plus indices `1,...,fireState.Length`, where `animIndex = 0` doesn't correspond to an entry in `fireState`), then the condition ought to be `animIndex > fireState.Length`.

Comment (SuicidalInsanity) [reply BDArmory/Weapons/ModuleWeapon.cs]: Ok, sitting down and going through it again, yeah, you are correct. Guess I wasn't thinking straight when I answered you last time. Updated PR with change

**Pull request merged**